[material-next] Drop the package

mnajdova opened the pull request:
We are dropping the `@mui/material-next` package. We are already working on the next Material UI version, which is about allowing static CSS extraction. We don't have the bandwidth to keep the focus on both of these efforts at the same time, and it is a maintenance burden to keep the package alive. 

Once we start to work on v7 and focus on the design-related updates, we will be directly updating the Material UI components, likely working on the next branch alongside the ongoing stable release.

DiegoAndai reviewed (commented):
- Will he latest released version of the package still be available to users?
- Should we keep the components' "Material Design 3 version" section? I think it's better to add a callout explaining that the `material-next` package is not being worked on while we wait to start working on v7. Simply removing it seems that it might scare people.

mnajdova commented:
> Will he latest released version of the package still be available to users?

Yes, I don't plan to remove it from npm now. We are discontinuing the work on it. 

> Should we keep the components' "Material Design 3 version" section? I think it's better to add a callout explaining that the material-next package is not being worked on while we wait to start working on v7. Simply removing it seems that it might scare people.

I think it may be misleading to keep the demos, we do not want people to keep experimenting with this package, that was the sole purpose for adding these. Based on the number of downloads I wouldn't worry: https://npm-stat.com/charts.html?package=%40mui%2Fmaterial-next&from=2023-03-17&to=2024-03-17

DiegoAndai commented:
> I think it may be misleading to keep the demos

Good point. Let's go ahead then.

We can leave a comment in #29345 explaining why those were removed.

> I wasn't sure if I should update the second part of https://mui.com/blog/2023-material-ui-v6-and-beyond/#material-ui-v7, where we mention the package. I left it for now.

I would update it, under the same argument that we're removing the demos.

mnajdova commented:
> > I wasn't sure if I should update the second part of https://mui.com/blog/2023-material-ui-v6-and-beyond/#material-ui-v7, where we mention the package. I left it for now.
> 
> I would update it, under the same argument that we're removing the demos.

Fair enough, updated and I also updated the all components page to not show the MD3 tags anywhere.

DiegoAndai commented:
> Fair enough, updated and I also updated the all components page to not show the MD3 tags anywhere.

cc: @danilo-leal

oliviertassinari commented:
It looks like https://www.npmjs.com/package/@mui/material-next should have been deprecated, but it wasn't. People got confused about it, e.g. https://x.com/michioxd/status/1873768653625368644.

Fixed: 
<img width="1196" alt="SCR-20241231-mzqy" src="https://github.com/user-attachments/assets/bda96dc6-b602-4acf-88b7-6e98470e95a3" />
